src/clj/rems/api/services/catalogue.clj

```diff
@@ -17,8 +17,21 @@
      :licenses (licenses/get-licenses {:wfid wfid
                                        :items [id]})}))
 
+(defn- organization-mismatch-error [{:keys [organization form resid wfid]}]
+  (let [mismatches (merge (when (not= organization (:form/organization (form/get-form-template form)))
+                            {:form form})
+                          (when (not= organization (:organization (resource/get-resource resid)))
+                            {:resource resid})
+                          (when (not= organization (:organization (workflow/get-workflow wfid)))
+                            {:workflow wfid}))]
+    (when (not (empty? mismatches))
```

Comment:
I'd prefer just `(when (seq mismatches) ...)` here. It's currently the most popular option in the codebase?

Reply:
somehow `empty?` feels more natural for maps. for vectors & lists `seq` is natural

Reply:
Hmm, is `(when-let [mismatches ...] ...)` possible?